lib/js/jsbigints.nim

```diff
@@ -0,0 +1,137 @@
+  ## Constructor for `JsBigInt`.
+  runnableExamples:
+    doAssert big"-1" == big"1" - big"2"
+
+func toLocaleString*(this: JsBigInt): cstring {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: cstring): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: openArray[cstring]): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func toString*(this: JsBigInt): cstring {.importjs: "#.toString()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string = $toString(this)
+  ## Return a string representation of `JsBigInt`.
+
+func asIntN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
+
+func asUintN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+
+func `+`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".}
+   ## Same as `div` but for `JsBigInt`.
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".}
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+
+func `+=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `-=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `*=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `div=`*(x, y: JsBigInt) {.importjs: "(# /= #)".}
+
+func `mod=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
+
+func `inc`*(x: JsBigInt) {.importjs: "(++#)".}
+  ## Same as `inc` but for `JsBigInt` (uses JavaScript `++BigInt()`).
+
+func `dec`*(x: JsBigInt) {.importjs: "(--#)".}
+  ## Same as `dec` but for `JsBigInt` (uses JavaScript `--BigInt()`).
+
+func `<`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".}
+
+func `<=`*(x, y: JsBigInt): bool {.importjs: "(# $1 #)".}
+
+func `==`*(x, y: JsBigInt): bool {.importjs: "(# === #)".}
+
+func `**`*(x, y: JsBigInt): JsBigInt {.importjs: "((#) $1 #)".}
+
+func `and`*(x, y: JsBigInt): JsBigInt {.importjs: "(# && #)".}
+
+func `or`*(x, y: JsBigInt): JsBigInt {.importjs: "(# || #)".}
+
+func `not`*(x: JsBigInt): JsBigInt {.importjs: "(!#)".}
+
+func `in`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
```

Comment:
I don't understand this; how do you use it? it would need a runnableExamples at least